x/virtualgroup/keeper/keeper.go

```diff
@@ -289,6 +294,8 @@ func (k Keeper) SwapAsPrimarySP(ctx sdk.Context, primarySP, successorSP *sptypes
 
 	srcStat := k.MustGetGVGStatisticsWithinSP(ctx, primarySP.Id)
 	dstStat := k.GetOrCreateGVGStatisticsWithinSP(ctx, successorSP.Id)
+	srcVGFStat := k.MustGetGVGFamilyStatisticsWithinSP(ctx, primarySP.Id)
```

Comment:
if this is executed before the hardfork, there is no stats yet.

Reply:
fixed